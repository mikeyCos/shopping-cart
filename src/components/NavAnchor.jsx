import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import styles from "../styles/NavAnchor.module.css";

const NavAnchor = ({ pathname }) => {
  return (
    <NavLink
      className={({ isActive, isPending, isTransitioning }) =>
        [
          isPending ? "pending" : "",
          isActive ? styles["active"] : "",
          isTransitioning ? "transitioning" : "",
        ]
          .filter((item) => item)
          .join(" ")
      }
      to={`/${pathname}`}
    >
      {pathname}
    </NavLink>
  );
};

NavAnchor.propTypes = {
  pathname: PropTypes.string.isRequired,
};

export default NavAnchor;
