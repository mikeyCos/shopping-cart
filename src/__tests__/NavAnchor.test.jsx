import { describe, expect, it, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavAnchor from "../components/NavAnchor";

describe("NavAnchor component", () => {
  it("NavAnchor component matches snapshot", () => {
    const pathname = "placeholder";
    const { container } = render(
      <MemoryRouter>
        <NavAnchor pathname={pathname} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it("NavAnchor textContent is 'home ", () => {
    const pathname = "home";
    render(
      <MemoryRouter>
        <NavAnchor pathname={pathname} />
      </MemoryRouter>
    );

    const anchor = screen.getByRole("link");
    expect(anchor.textContent).toMatch(/home/i);
  });
});
