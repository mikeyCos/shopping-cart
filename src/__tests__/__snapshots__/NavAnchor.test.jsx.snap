// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavAnchor component > NavAnchor component matches snapshot 1`] = `
<div>
  <a
    class=""
    href="/placeholder"
  />
</div>
`;
